ReactAndroid/src/main/java/com/facebook/react/uimanager/TransformHelper.java

```diff
@@ -0,0 +1,91 @@
+package com.facebook.react.uimanager;
+
+import com.facebook.react.bridge.JSApplicationIllegalArgumentException;
+import com.facebook.react.bridge.ReadableArray;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.ReadableType;
+
+/**
+ */
+public class TransformHelper {
+
+  private static final double[] HELPER_MATRIX = new double[16];
```

Comment:
Can we actually make this a ThreadLocal? I'm concerned that we will want to do this math both on the shadow and main threads at some point and this isn't currently threadsafe.